glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -363,14 +372,48 @@ fn seg_seg_intersection(seg1: PathSeg, seg2: PathSeg) -> Option<Intersection> {
                 t0: hit.segment_t,
                 t1: hit.line_t,
             }),
-        (bez0, bez1) => curve_curve_intersection_py(bez0, bez1),
+        (bez0, bez1) => return curve_curve_intersection_py(bez0, bez1),
+    }?;
+    if let (PathSeg::Line(l1), PathSeg::Line(l2)) = (seg1, seg2) {
+        let pt = l1.eval(hit.t0);
+        // the bezierTools code for line intersections has a bunch of special
+        // cases that were causing us to deviate, so we try to cover those here
+        // as they come up:
+        //
+        // special check for close x coords:
+        // https://github.com/fonttools/fonttools/blob/a6f59a4f87a011/Lib/fontTools/misc/bezierTools.py#L1193-L1212
+
+        if py_isclose(l1.end().x, l1.start().x) || py_isclose(l2.start().x, l2.end().x) {
+            return Some(hit);
+        }
+        // final guard statement
+        // https://github.com/fonttools/fonttools/blob/a6f59a4f87a/Lib/fontTools/misc/bezierTools.py#L1221-L1223
+        //eprintln!("POINT {pt} {l1:?}, {l2:?}");
+        if !(both_points_are_on_same_side_of_origin(pt, l1.p1, l1.p0)
+            && both_points_are_on_same_side_of_origin(pt, l2.p0, l2.p1))
```

Comment:
`both_points...` followed by 3 point args threw me at first, I suggest a trait here too. `l1.on_one_side_of(pt)` perhaps.

Reply:
this is also me using the same name and arguments as the python fn referenced above. (there's a stray comment here tho, will clean that up)

Reply:
A trait with a comment would be easier to read and still equally linked to python?